src/dsl/validators.ts

```diff
@@ -0,0 +1,93 @@
+import Intructor, { type OAIClientExtended } from "@/instructor"
+import OpenAI from "openai"
+import * as z from "zod"
+
+const Validator = z.object({
+  valid: z
+    .boolean()
+    .default(true)
+    .describe("Whether the attribute is valid based on the requirements"),
+  reason: z
+    .string()
+    .optional()
+    .describe("The error message if the attribute is not valid, otherwise None"),
+  suggestedValue: z
+    .string()
+    .optional()
+    .describe("If the attribute is not valid, suggest a new value for the attribute")
+})
+
+export function llmValidator({
+  statement,
+  allowOverride = false,
+  model = "gpt-3.5-turbo",
+  temperature = 0,
+  openaiClient
+}: {
+  statement: string
+  allowOverride?: boolean
+  model?: string
+  temperature?: number
+  openaiClient?: OAIClientExtended
+}) {
+  const client = openaiClient ?? Intructor({ client: new OpenAI(), mode: "FUNCTIONS" })
+
+  //eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async function llm(v: any, ctx: z.RefinementCtx) {
+    const response = await client.chat.completions.create({
+      response_model: Validator,
+      messages: [
+        {
+          role: "system",
+          content: `# MISSION
+You are a world class validation model. Capable to determine if the value provided is valid based on the statement given. If it is not, give a reason why and suggest a new value.
```

Comment:
mission?

Reply:
Updated the prompt to try to get some of the tests working properly. It's based off of these templates that I found 
https://github.com/daveshap/ChatGPT_Custom_Instructions/blob/main/academic_copy_reviser.md